ui/number-field.reel/number-field.js

```diff
@@ -1,40 +1,49 @@
 /**
-    @module "montage/ui/native/input-number.reel"
-    @requires montage/ui/component
+    @module "montage/ui/number-field.reel"
     @requires montage/ui/text-input
 */
 
-var TextInput = require("ui/text-input").TextInput;
-
+var TextInput = require("ui/text-input").TextInput,
+    MinMaxConverter = require("core/converter/min-max-converter.js").MinMaxConverter;
+   
 /**
  * Wraps the a &lt;input type="date"> element with binding support for the element's standard attributes.
    @class module:"montage/ui/native/input-number.reel".InputNumber
    @extends module:montage/ui/text-input.TextInput
  */
 var NumberField = exports.NumberField = TextInput.specialize({
-	hasTemplate: {value: false }
+
+    enterDocument:{
+        value: function() {
+            if(this.min || this.max) {
```

Comment:
What if either this.min or this.max is not a number/undefined? We should add something like 
this.converter.min = this.min || Infinity;
this.converter.max = this.max || Infinity;